Sources/Diff/FullDiff.php

```diff
@@ -1518,9 +1518,10 @@ protected static function redistributeContextLines(array $changes): array
 	 *
 	 * @param array $where Info about which change parts have no final newline.
 	 * @param array $changes The diff's complete set of changes.
+	 * @param string $section Which section
 	 * @return array Updated $changes.
 	 */
-	protected static function fixFinalLineEndings(array $where, array $changes): array
+	protected static function fixFinalLineEndings(array $where, array $changes, string $section): array
```

Comment:
This is not the right way to fix this. The problem is simply that all occurrences of `$section` inside this function should actually be `$part`.